Rules/UseLiteralInitializerForHashtable.cs

```diff
@@ -0,0 +1,401 @@
+                return AstVisitAction.SkipChildren;
+            }
+
+            var commandName = commandAst.GetCommandName();
+            if (commandName == null
+                || !commandName.Equals("new-object", StringComparison.OrdinalIgnoreCase))
+            {
+                return AstVisitAction.Continue;
+            }
+
+            AnalyzeNewObjectCommand(commandAst);
+            return AstVisitAction.Continue;
+        }
+
+        /// <summary>
+        /// Checks if a hashtable is created using [hashtable]::new()
+        /// </summary>
+        public override AstVisitAction VisitInvokeMemberExpression(InvokeMemberExpressionAst methodCallAst)
+        {
+            if (methodCallAst == null)
+            {
+                return AstVisitAction.SkipChildren;
+            }
+
+            var typeExprAst = methodCallAst.Expression as TypeExpressionAst;
+            if (typeExprAst == null
+                || !presetTypeNameSet.Contains(typeExprAst.TypeName.FullName))
+            {
+                return AstVisitAction.Continue;
+            }
+
+            var memberStringConstantExprAst = methodCallAst.Member as StringConstantExpressionAst;
+            if (memberStringConstantExprAst == null
+                || !memberStringConstantExprAst.Value.Equals("new", StringComparison.OrdinalIgnoreCase))
+            {
+                return AstVisitAction.Continue;
+            }
+
+            // no arguments provided to new OR one of the argument ends with ignorecase
+            // (heuristics find to something like [system.stringcomparer]::ordinalignorecase)
+            if (methodCallAst.Arguments == null
+                || !HasIgnoreCaseComparerArg(methodCallAst.Arguments))
+            {
+                var dr = new DiagnosticRecord(
+                    Strings.UseLiteralInitilializerForHashtableDescription,
+                    methodCallAst.Extent,
+                    GetName(),
+                    GetDiagnosticSeverity(),
+                    fileName,
+                    ruleId: null,
+                    suggestedCorrections: GetSuggestedCorrections(methodCallAst, this.fileName));
+                diagnosticRecords.Add(dr);
+            }
+
+            return AstVisitAction.Continue;
+        }
+
+        /// <summary>
+        /// Analyzes command ast to check for new-object command and parse its arguments
+        /// </summary>
+        private void AnalyzeNewObjectCommand(CommandAst commandAst)
+        {
+            string typeName;
+            List<string> argumentList;
+            GetParametersFromCommandAst(commandAst, out typeName, out argumentList);
+            if (typeName == null
+                || !presetTypeNameSet.Contains(typeName))
+            {
+                return;
+            }
+
+            if (argumentList != null)
+            {
+                if (argumentList.Any(arg => arg != null && arg.EndsWith("ignorecase", StringComparison.OrdinalIgnoreCase)))
+                {
```

Comment:
You should probably be more selective than this - a hashtable uses `StringComparer.CurrentCultureIgnoreCase` - so you wouldn't want to ignore `[hashtable]::new(0, [stringcomparer]::OrdinalIgnoreCase)`